app/src/main/java/com/github/se/polyfit/MainActivity.kt

```diff
@@ -75,6 +76,10 @@ class MainActivity : ComponentActivity() {
                 navigateBack = navigation::goBack, navigateTo = navigation::navigateToAddMeal)
           }
           composable(Route.AddMeal) { AddMealFlow(navigation::goBack, navigation::navigateToHome) }
+
+          composable(Route.RecipeRecommendation) {
+            RecipeRecommendationScreen(navController = navController)
```

Comment:
I completely agree with you. However, since I am using the pre-implemented component `GenericScreen.kt`, which directly uses the `navController`, I can't do much without refactoring this component. This should be done in another PR. If you have some free time, feel free to refactor it.